tensorflow_datasets/image/icoseg.py

```diff
@@ -0,0 +1,76 @@
+
+"""
+CMU-Cornell iCoseg dataset
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@inproceedings {
+title = {iCoseg: Interactive Co-segmentation with Intelligent Scribble Guidance},
+author = {D. Batra and A. Kowdle and D. Parikh and J. Luo and T. Chen},
+booktitle = {ECCV},
+year = {2010} }
+"""
+
+_DESCRIPTION = """\
+This is the CMU-Cornell iCoseg dataset for image segmentation. The dataset contains 38 different \
+groups totalling 643 images. These images also have there respective pixel level groundtruth \
+annotations. These groundtruths segment the images into two classes - the subject and the background.
+"""
+
+
+class Icoseg(tfds.core.GeneratorBasedBuilder):
+  """iCoseg dataset for segmentation."""
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+      builder=self,
+      description=_DESCRIPTION,
+      features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(shape=(None, None, 3), encoding_format="jpeg"),
+        "label_png": tfds.features.Image(shape=(None, None, 1), encoding_format="png"),
+      }),
+      supervised_keys=("image", "label_png"),
+      homepage="http://chenlab.ece.cornell.edu/projects/touch-coseg/",
+      citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    extracted_dir = dl_manager.download_and_extract({
+      "iCoseg_dir_path": "http://chenlab.ece.cornell.edu/projects/touch-coseg/CMU_Cornell_iCoseg_dataset.zip"
+    })
+    return [
+      tfds.core.SplitGenerator(
+        name=tfds.Split.TRAIN,
```

Comment:
What about the TEST and VALIDATION splits?

Reply:
@mrinalTheCoder most likely they are not specified in the dataset

Reply:
Yeah, the dataset does not specify the splits.